rkt/metadata_service_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2014 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/coreos/rocket/Godeps/_workspace/src/github.com/appc/spec/schema"
+	"github.com/coreos/rocket/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+)
+
+func setupPodStoreTest(t *testing.T) (*podStore, *types.UUID, string, string) {
+	ps := newPodStore()
+
+	uuid, err := types.NewUUID("de305d54-75b4-431b-adb2-eb6b9e546013")
+	if err != nil {
+		t.Fatalf("bad uuid literal")
+	}
+
+	ip := "1.2.3.4"
+	app := "myapp"
+
+	pm := &schema.PodManifest{}
+	ps.addPod(uuid, ip, pm)
+
+	im := &schema.ImageManifest{}
+	err = ps.addApp(uuid, app, im)
+	if err != nil {
+		t.Fatalf("addApp failed with %v", err)
+	}
+
+	return ps, uuid, ip, app
+}
+
+func TestPodStoreAddApp(t *testing.T) {
+	ps, _, _, app := setupPodStoreTest(t)
+
+	uuid2, err := types.NewUUID("fe305d54-75b4-431b-adb2-eb6b9e546013")
+	if err != nil {
+		t.Fatalf("bad uuid literal")
```

Comment:
I think it is a useful distinction here, panic => probably typed the wrong UUID, never expected; error/fatal => test behaviour failed